routes/categories.router.js

```diff
@@ -0,0 +1,47 @@
+const express = require("express");
+const router = express.Router();
+const Category = require("../model/categories.model");
+const mongoose = require("mongoose");
+
+router.get("/", (req, res) => {
+	Category.find({})
+		.populate("products")
+		.select("-__v")
+		.then((data) => {
+			res.json({ categories: data });
+		})
+		.catch((error) => {
+			console.error(error);
+			res.json({ message: "Some error occurred!" });
+		});
+});
+
+router.post("/", (req, res) => {
+	const category = req.body;
+	const newCategory = new Category(category);
+	newCategory
+		.save()
+		.then((data) => {
+			res.status(201).json({ category: data });
+		})
+		.catch((error) => {
+			console.error(error);
+			res.status(400).json({ message: "Validation Failed" });
+		});
+});
+
+router.get("/:categoryId", (req, res) => {
+	const { categoryId } = req.params;
+	Category.findById({ _id: categoryId })
```

Comment:
This code can be seperated out into seperate function like paramHandlers 
router.param("categoryId",categoryParamHandler) So it can be reused wherever required

Reply:
I feel like writing directly here makes it more readable, also I don't have many param handlers so I have directly written it here.